app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -546,6 +546,154 @@ class StateFragmentTest {
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_landscape_thirdState_submitAnswer_submitButtonIsEnabled() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(matches(isEnabled()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_land_thirdState_submitAnswer_clickSubmit_continueIsVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
+      onView(withId(R.id.continue_navigation_button)).check(
+        matches(withText(R.string.state_continue_button))
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_submitInvalidAnswer_disablesSubmitAndShowsError() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      // Attempt to submit an invalid answer.
+      selectMultipleChoiceOption(optionPosition = 1, expectedOptionText = "Chicken")
+      clickSubmitAnswerButton()
+
+      // The submission button should now be disabled and there should be an error.
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_land_thirdState_submitInvalidAnswer_disablesSubmitAndShowsError() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      // Attempt to submit an invalid answer.
+      selectMultipleChoiceOption(optionPosition = 1, expectedOptionText = "Chicken")
+      clickSubmitAnswerButton()
+
+      // The submission button should now be disabled and there should be an error.
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_invalidAnswer_updated_submitAnswerIsEnabled() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      // Attempt to submit an invalid answer.
```

Comment:
```suggestion
      // Attempt to submit an invalid answer.
```

Nit: This is part of the arrangement.

Reply:
updated